samples/subsys/usb/console/src/main.c

```diff
@@ -6,9 +6,19 @@
 
 #include <zephyr.h>
 #include <misc/printk.h>
+#include <misc/util.h>
+#include <string.h>
 
 void main(void)
 {
+	if (strncmp(CONFIG_UART_CONSOLE_ON_DEV_NAME, CONFIG_CDC_ACM_PORT_NAME,
+		    min(strlen(CONFIG_UART_CONSOLE_ON_DEV_NAME),
+			strlen(CONFIG_CDC_ACM_PORT_NAME)))) {
+		printk("Error: Console device name is not USB ACM\n");
+
```

Comment:
I guess you want to first check that both lengths are equal, and then compare the strings, otherwise `foobar0` and `foobar` will match.

In addition to that, both `CONFIG_UART_CONSOLE_ON_DEV_NAME` and `CONFIG_CDC_ACM_PORT_NAME` are fixed values, so Coverity will report a `Pointless string comparison` issue. I don't have a solution for that.

Reply:
Added length comparison also.